src/Avalonia.Input/MouseDevice.cs

```diff
@@ -435,137 +397,15 @@ PointerPointProperties CreateProperties(RawPointerEventArgs args)
-            if(clearRoot)
-            {
-                e.Source = element;
-                e.Handled = false;
-                element.RaiseEvent(e);
-            }
-        }
-
-        private IInputElement? SetPointerOver(IPointerDevice device, ulong timestamp, IInputRoot root, Point p, 
-            PointerPointProperties properties,
-            KeyModifiers inputModifiers)
-        {
-            device = device ?? throw new ArgumentNullException(nameof(device));
-            root = root ?? throw new ArgumentNullException(nameof(root));
-
-            var element = root.InputHitTest(p);
-
-            if (element != root.PointerOverElement)
-            {
-                if (element != null)
-                {
-                    SetPointerOver(device, timestamp, root, element, properties, inputModifiers);
-                }
-                else
-                {
-                    ClearPointerOver(device, timestamp, root, properties, inputModifiers);
-                }
-            }
-
-            return element;
-        }
-
-        private void SetPointerOver(IPointerDevice device, ulong timestamp, IInputRoot root, IInputElement element,
-            PointerPointProperties properties,
-            KeyModifiers inputModifiers)
-        {
-            device = device ?? throw new ArgumentNullException(nameof(device));
-            root = root ?? throw new ArgumentNullException(nameof(root));
-            element = element ?? throw new ArgumentNullException(nameof(element));
-
-            IInputElement? branch = null;
-
-            IInputElement? el = element;
-
-            while (el != null)
-            {
-                if (el.IsPointerOver)
-                {
-                    branch = el;
-                    break;
-                }
-                el = (IInputElement?)el.VisualParent;
-            }
-
-            el = root.PointerOverElement;
-
-            var e = CreateSimpleEvent(InputElement.PointerLeaveEvent, timestamp, el, properties, inputModifiers);
-            if (el!=null && branch!=null && !el.IsAttachedToVisualTree)
-            {
-                ClearChildrenPointerOver(e,branch,false);
-            }
-            
-            while (el != null && el != branch)
-            {
-                e.Source = el;
-                e.Handled = false;
-                el.RaiseEvent(e);
-                el = (IInputElement?)el.VisualParent;
-            }            
-
-            el = root.PointerOverElement = element;
-            e.RoutedEvent = InputElement.PointerEnterEvent;
-
-            while (el != null && el != branch)
-            {
-                e.Source = el;
-                e.Handled = false;
-                el.RaiseEvent(e);
-                el = (IInputElement?)el.VisualParent;
-            }
-        }
-
         public void Dispose()
         {
             _disposed = true;
             _pointer?.Dispose();
         }
+
+        public void TopLevelClosed(IInputRoot root)
```

Comment:
Left for compatibility, should be removed with following PR with breaking changes.